tests/test_energy_overflows.py

```diff
@@ -272,13 +272,22 @@ def test_energy_overflows_with_summation_of_energies(precision):
     )
     nonbonded_gpu = nonbonded.to_gpu(precision)
 
-    # TBD: Remove the try/finally to clean up the GPU potential, else calling reset_device
-    # will fail
-    try:
-        # The reference should return a large energy value
-        assert nonbonded(x, nb_params, box) > 0.0
+    # The reference should return a value larger than we can express in fixed point
+    assert nonbonded(x, nb_params, box) > np.iinfo(np.int64).max / FIXED_EXPONENT
 
-        # The GPU potentials will overflow, resulting in a nan value
-        assert not np.isfinite(nonbonded_gpu(x, nb_params, box))
-    finally:
-        del nonbonded_gpu
+    # The GPU potentials will overflow, resulting in a nan value
+    assert np.isnan(nonbonded_gpu(x, nb_params, box))
+
+
+def test_energy_accumulation():
+    """Test the the logic used to accumulate energy in int128.
+
+    It relies on doing a block level parallel reduce for performance.
+    """
+    rng = np.random.default_rng(2023)
+
+    vals = rng.integers(-10000, 10000, size=10000, dtype=np.int64)
```

Comment:
Might be good to test some different values of size as well?  (eg. 0, <32/warp, < threads per block, etc.)

Reply:
Added additional tests in [d59a706](https://github.com/proteneer/timemachine/pull/1090/commits/d59a70604e18418c9875dda5e1fb27250f196f60)